src/hub/action_form.ts

```diff
@@ -14,7 +20,8 @@ export interface ActionFormField {
   label?: string
   description?: string
   default?: string
-  type?: "string" | "textarea" | "select"
+  type?: "string" | "textarea" | "select" | "oauth_link"
```

Comment:
If these are only respected when type has a particular value, we should encode this into the ActionFormField type so the type system will prevent you from specifying them on a string.

Reply:
This will make the TS compiler errors enforce the required properties for each `type` of field.

```ts
export interface ActionFormFieldBase {
  name: string
  label?: string
  description?: string
  default?: string
  required?: boolean
}

export interface ActionFormFieldString extends ActionFormFieldBase {
    type?: "string" | "textarea"
}

export interface ActionFormFieldSelect extends ActionFormFieldBase {
    type: "select" 
    options: { name: string, label: string }[]
}

export interface ActionFormFieldOAuth extends ActionFormFieldBase {
  type: "oauth_link" 
  oauth_url: string
}

export type ActionFormField = ActionFormFieldString | ActionFormFieldSelect | ActionFormFieldOAuth
```